### PR TITLE
 Add file support to AnonymousApexTask

### PR DESCRIPTION
**Pull request by davisagli**
We should be able to replace the similar tasks in SAL and Cumulus with this.

**lurch: attach W-018728

### CODE REVIEW

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000006POTmAAO" target="_blank">W-018728</a>

File: cumulusci/tasks/apex/anon.py [hunk 1]
```diff
@@ -1,35 +1,110 @@
...
         )
-        if result.status_code != 200:
-            raise SalesforceGeneralError(url, path, result.status_code, result.content)
+        self._check_result(result)
+
+        self.logger.info("Anonymous Apex Success")
+
+    def _prepare_apex(self, apex):
+        # Replace the email string only if it exists
+        admin_email = self.project_config.dev_config__admin_email
```
Comment (cdcarter) [inline cumulusci/tasks/apex/anon.py]: can we add this convention to the docs somewhere?

File: cumulusci/tasks/apex/anon.py [hunk 2]
```diff
@@ -1,35 +1,110 @@
...
+        managed = self.options.get('managed') or False
+        namespaced = self.options.get('namespaced') or False
+        namespace = self.project_config.project__package__namespace
+        namespace_prefix = ''
+        record_type_prefix = ''
+        if managed or namespaced:
+            namespace_prefix = namespace + '__'
+        if namespaced:
+            record_type_prefix = namespace + '.'
+        apex = apex.replace('%%%NAMESPACE%%%', namespace_prefix)
```
Comment (cdcarter) [inline cumulusci/tasks/apex/anon.py]: run `black`

File: cumulusci/tasks/apex/anon.py [hunk 3]
```diff
@@ -1,35 +1,110 @@
...
+        apex = apex.replace('%%%NAMESPACED_ORG%%%', namespace_prefix)
+        apex = apex.replace('%%%NAMESPACED_RT%%%', record_type_prefix)
```
Comment (cdcarter) [inline cumulusci/tasks/apex/anon.py]: other tasks take these tokens as options. we should probably do the same here, or consider adding these to the project_config. very magic stringy rn. but, a backlog item and not required for merge unless you want to do it now.

File: cumulusci/tasks/apex/anon.py (hunk 2 above)
Comment (davisagli) [reply cumulusci/tasks/apex/anon.py]: Okay, actually got it set up to format on save now.

File: cumulusci/tasks/apex/anon.py (hunk 3 above)
Comment (davisagli) [reply cumulusci/tasks/apex/anon.py]: I think I'll leave it until there's an explicit need. This is just copied over from what's in SAL and Cumulus.

File: cumulusci/tasks/apex/anon.py (hunk 1 above)
Comment (cdcarter) [reply cumulusci/tasks/apex/anon.py]: a new block in `features.rst` called Anonymous Apex that explains this email behavior, as well as the concat behavior and how it is used to call functions of a shared file?

Comment (davisagli) [reply cumulusci/tasks/apex/anon.py]: I made the task doc generator include extra docs from a `task_docs` attribute on the class, and added a few pointers there. Sorry for putting this in the same commit that includes running black...

**Pull request merged**